File: clients/PYTHON-client/benchmark.py
import time
import sys
import os

# Add parent dir to path to find solidb
sys.path.append(os.path.dirname(os.path.abspath(__file__)))
from solidb import Client

def run_benchmark():
    port = int(os.environ.get("SOLIDB_PORT", "9998"))
    password = os.environ.get("SOLIDB_PASSWORD", "password")
    
    client = Client("127.0.0.1", port)
    client.connect()
    client.auth("_system", "admin", password)
    
    db = "bench_db"
    col = "python_bench"
    
    try:
        client.create_database(db)
    except: pass
    
    try:
        client.create_collection(db, col)
    except: pass

    iterations = 1000
    
    # INSERT BENCHMARK
    start_time = time.time()
    for i in range(iterations):
        key = f"bench_{i}"
        client.insert(db, col, {"id": i, "data": "benchmark data content"}, key=key)
    end_time = time.time()
    
    insert_duration = end_time - start_time
    insert_ops_per_sec = iterations / insert_duration
    print(f"PYTHON_BENCH_RESULT:{insert_ops_per_sec:.2f}")
    
    # READ BENCHMARK
    start_time = time.time()
    for i in range(iterations):
        key = f"bench_{i}"
        client.get(db, col, key)
    end_time = time.time()
    
    read_duration = end_time - start_time
    read_ops_per_sec = iterations / read_duration
    print(f"PYTHON_READ_BENCH_RESULT:{read_ops_per_sec:.2f}")
    
    client.close()

if __name__ == "__main__":
    run_benchmark()

File: clients/Ruby-client/benchmark.rb
require_relative 'lib/solidb/client'
require 'time'

def run_benchmark
  port = (ENV['SOLIDB_PORT'] || '9998').to_i
  password = ENV['SOLIDB_PASSWORD'] || 'password'
  
  client = SoliDB::Client.new('127.0.0.1', port)
  client.connect
  client.auth('_system', 'admin', password)
  
  db = 'bench_db'
  col = 'ruby_bench'
  
  begin
    client.create_database(db)
  rescue => e
  end
  
  begin
    client.create_collection(db, col)
  rescue => e
  end

  iterations = 1000
  
  start_time = Time.now
  iterations.times do |i|
    client.insert(db, col, { id: i, data: "benchmark data content" })
  end
  end_time = Time.now
  
  duration = end_time - start_time
  ops_per_sec = iterations / duration
  
  puts "RUBY_BENCH_RESULT:#{ops_per_sec.round(2)}"
  client.close
end

run_benchmark
